apps/meteor/client/views/room/composer/messageBox/MessageBox.tsx

```diff
@@ -158,7 +216,164 @@ const MessageBox = ({
 		chat.emojiPicker.open(ref, (emoji: string) => chat.composer?.insertText(` :${emoji}: `));
 	});
 
-	const handleSendMessage = useMutableCallback(() => {
+	const handleSendMessage = useMutableCallback(async () => {
+		if (uploadfiles !== undefined && uploadfiles.length > 0) {
+			const msg = chat.composer?.text ?? '';
+			if (uploadfiles.length > 6) {
+				dispatchToastMessage({
+					type: 'error',
+					message: "You can't upload more than 6 files at once",
+				});
+				chat.composer?.clear();
+				setUploadfiles([]);
+				return;
+			}
+			for (const queuedFile of uploadfiles) {
+				const { name, size } = queuedFile;
+				if (!name) {
+					dispatchToastMessage({
+						type: 'error',
+						message: t('error-the-field-is-required', { field: t('Name') }),
+					});
+					chat.composer?.clear();
+					setUploadfiles([]);
+					return;
+				}
+
+				if (maxFileSize > -1 && (size || 0) > maxFileSize) {
+					dispatchToastMessage({
+						type: 'error',
+						message: `${t('File_exceeds_allowed_size_of_bytes', { size: fileSize(maxFileSize) })}`,
+					});
+					chat.composer?.clear();
+					setUploadfiles([]);
+					return;
+				}
+			}
+
+			Object.defineProperty(uploadfiles[0], 'name', {
+				writable: true,
+				value: uploadfiles[0].name,
+			});
+
+			const e2eRoom = await e2e.getInstanceByRoomId(room._id);
```

Comment:
Please move the whole logic below to the upload function, or create a new upload function that handles all this e2ee thing and call the final upload function internaly, this logic should not be inside the composer component, the upload may happen from other places

Reply:
done sir i have made completely new function to handleFileUploads and also made one more new function to handle e2ee things